src/Modera/SecurityBundle/PasswordStrength/SemanticPasswordConfig.php

```diff
@@ -40,9 +40,23 @@ public function isNumberRequired()
     /**
      * {@inheritdoc}
      */
-    public function isCapitalLetterRequired()
+    public function isLetterRequired()
     {
-        return $this->semanticConfig['capital_letter_required'];
+        return false !== $this->semanticConfig['letter_required'];
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getLetterRequiredType()
+    {
+        $available = array('capital_or_non_capital', 'capital_and_non_capital', 'capital', 'non_capital');
+
+        if (!in_array($this->semanticConfig['letter_required'], $available)) {
```

Comment:
We should avoid having validation performed here, it is just an abstraction of where the config comes from, see my other comment in Configuration class, you won't need this validation block here once you have refactored the code in Configuration class.